### PR TITLE
HIVE-26510: Minor compaction creates empty delta files when no prior …

### PR DESCRIPTION
**Pull request by SourabhBadhya**
…delta file exists

### What changes were proposed in this pull request?

Minor compaction creates empty delta files when no prior delta file exists. This patch tries to fix by not creating the empty delta files.

### Why are the changes needed?

Unnecessary empty delta files are created. This should be avoided.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit test

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java [hunk 1]
```diff
@@ -817,6 +816,7 @@ public void map(WritableComparable key, CompactorInputSplit split,
           deleteEventWriter.write(value);
           reporter.progress();
         } else {
+          getWriter(reporter, reader.getObjectInspector(), split.getBucket());
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java:819]: could we refactor to:
````
        if (sawDeleteRecord && acidOperationalProperties.isSplitUpdate()) {
            getDeleteEventWriter(reporter, reader.getObjectInspector(), split.getBucket())
                .write(value);
          reporter.progress();
        } else {
          getWriter(reporter, reader.getObjectInspector(), split.getBucket())
              .write(value);
          reporter.progress();
        }
````

Comment (SourabhBadhya) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java:819]: `getWriter` and `getDeleteEventWriter` methods initializes `writer` object and `deleteEventWriter` object respectively which are initially null.

Comment (SourabhBadhya) [reply ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactorMR.java]: Refactored. Done.

**Review by deniskuzZ — APPROVED** [review]
LGTM, pending tests

**Review by ayushtkn — APPROVED** [review]
LGTM

**Pull request merged**